src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -470,23 +626,52 @@ protected OriginEntryFull buildOffsetOriginEntry(OriginEntryFull baseEntry, Kual
     protected OriginEntryFull buildOriginEntry(ConcurStandardAccountingExtractDetailLine detailLine, KualiDecimal amount) {
         if (collectorHelper.isCashAdvanceLine(detailLine)) {
             return buildCashAdvanceOriginEntry(detailLine, amount);
+        } else if (collectorHelper.isAtmFeeDebitLine(detailLine)) {
+            return buildAtmFeeDebitOriginEntry(detailLine, amount);
         } else {
             return buildRegularOriginEntry(detailLine, amount);
         }
     }
 
-    protected OriginEntryFull buildCashAdvanceOriginEntry(ConcurStandardAccountingExtractDetailLine detailLine, KualiDecimal amount) {
-        OriginEntryFull originEntry = new OriginEntryFull();
-        ConcurRequestedCashAdvance requestedCashAdvance = requestedCashAdvancesByCashAdvanceKey.get(detailLine.getCashAdvanceKey());
-        if (requestedCashAdvance == null) {
-            throw new IllegalStateException("Did not find cash advance when building Collector line; this should NEVER happen. Key: "
-                    + detailLine.getCashAdvanceKey());
+    protected OriginEntryFull buildCashAdvanceOriginEntry(ConcurStandardAccountingExtractDetailLine detailLine,
+            KualiDecimal amount) {
+        OriginEntryFull originEntry;
+        if (collectorHelper.isAtmCashAdvanceLine(detailLine)) {
+            originEntry = buildOriginEntryFromDetailLineForAtmCashAdvance(detailLine);
+        } else {
+            ConcurRequestedCashAdvance requestedCashAdvance = requestedCashAdvancesByCashAdvanceKey.get(detailLine.getCashAdvanceKey());
+            if (requestedCashAdvance == null) {
+                throw new IllegalStateException(

```

Comment:
Do you think it would be helpful to add a LOG.error here?

Reply:
Farther up the call stack in the ConcurStandardAccountingExtractCollectorBatchBuilder class, there's a try-catch block that will log exceptions that are thrown by this class. Thus, I don't think an extra log statement is needed at this level.